src/hidimstat/noise_std.py

```diff
@@ -5,283 +5,256 @@
+                    + " noise assumption."
+                )
+        else:
+            raise ValueError("Unknown method for estimating the covariance matrix")
+        ## compute emperical correlation of the residual
+        if stationary:
+            # consideration of stationary noise
+            # (section 2.5 of `chevalier2020statistical`)
+            sigma_hat = np.median(sigma_hat_raw) * np.ones(n_times)
+            # compute rho from the empirical correlation matrix
+            # (section 2.5 of `chevalier2020statistical`)
+            corr_emp = np.corrcoef(residual.T)
+        else:
+            sigma_hat = sigma_hat_raw
+            residual_rescaled = residual / sigma_hat
+            corr_emp = np.corrcoef(residual_rescaled.T)
+
+        # TODO: Why the name of the method is different
+        #           than the name of the "function"?
+        # Median method
+        if not stationary or method == "simple":
+            rho_hat = np.median(np.diag(corr_emp, 1))
+            # estimate M (section 2.5 of `chevalier2020statistical`)
+            corr_hat = toeplitz(np.geomspace(1, rho_hat ** (n_times - 1), n_times))
+            cov_hat = np.outer(sigma_hat, sigma_hat) * corr_hat
+
+        # Yule-Walker method (algorithm in section 3 of `eshel2003yule`)
+        elif stationary and method == "AR":
+            # compute the autocorrelation coefficients of the AR model
+            rho_ar = np.zeros(order + 1)
+            rho_ar[0] = 1
+
+            for i in range(1, order + 1):
+                rho_ar[i] = np.median(np.diag(corr_emp, i))
+
+            # solve the Yule-Walker equations (see eq.2 in `eshel2003yule`)
+            R = toeplitz(rho_ar[:-1])
+            coef_ar = solve(R, rho_ar[1:])
+
+            # estimate the variance of the noise from the AR model
+            residual_estimate = np.zeros((n_samples, n_times - order))
+            for i in range(order):
+                # time window used to estimate the residual from AR model
+                start = order - i - 1
+                end = -i - 1
+                residual_estimate += coef_ar[i] * residual[:, start:end]
+            residual_diff = residual[:, order:] - residual_estimate
+            sigma_eps = np.median(norm(residual_diff, axis=0) / np.sqrt(n_samples))
+
+            # estimation of the autocorrelation matrices
+            rho_ar_full = np.zeros(n_times)
+            rho_ar_full[: rho_ar.size] = rho_ar
+            for i in range(order + 1, n_times):
+                start = i - order
+                end = i
+                rho_ar_full[i] = np.dot(coef_ar[::-1], rho_ar_full[start:end])
+            corr_hat = toeplitz(rho_ar_full)
+
+            # estimation of the variance of an AR process
+            # from wikipedia it should be:
+            # VAR(X_t)=\frac{\sigma_\epsilon^2}{1-\phi^2}
+            # TODO there is a short difference between the code
+            # and the above formula
```

Comment:
Should I remove the comments above or not?

Reply:
This would deserve a small study. I'd rather remove the comment here and open an issue.

Reply:
See issue #164.